public/app/plugins/datasource/loki/LogContextProvider.ts

```diff
@@ -218,9 +208,25 @@ export class LogContextProvider {
     return expr;
   };
 
-  getInitContextFiltersFromLabels = async (labels: Labels) => {
-    await this.datasource.languageProvider.start();
-    const allLabels = this.datasource.languageProvider.getLabelKeys();
+  getInitContextFiltersFromLabels = async (labels: Labels, query?: LokiQuery) => {
+    if (!query || isEmpty(labels)) {
+      return [];
+    }
+
+    let allLabels: string[] = [];
+    if (isQueryWithParser(query.expr).parserCount !== 1) {
```

Comment:
I think it would be even more save if we turn this around, or just check for `isQueryWithParser(query.expr).parserCount !== 0`. Because for those edge cases with two parsers we would now again use the extracted label in the context query, wouldn't we?

Reply:
Updated to https://github.com/grafana/grafana/pull/67112/commits/4b88d6782ca2be4cc7c9ac52a41ccff7d9ca7032 - so we use `getLabelKeys` if there is no parser and `fetchSeriesLabels` if there is some parser (could be 1,2,3...10000).